programs/stake/src/rewards.rs

```diff
@@ -610,6 +612,29 @@ mod tests {
         );
     }
 
+    #[test_case(u64::MAX, 1_000, u64::MAX => panics "Rewards intermittent calculation should fit within u128")]
+    #[test_case(1, u64::MAX, u64::MAX => panics "Rewards should fit within u64")]
+    fn calculate_rewards_tests(stake: u64, rewards: u64, credits: u64) {
+        let mut vote_state = VoteState::default();
+
+        let stake = new_stake(stake, &Pubkey::default(), &vote_state, u64::MAX);
+
+        vote_state.increment_credits(0, credits);
+
+        calculate_stake_rewards(
+            0,
+            &stake,
+            &PointValue {
+                rewards: rewards,
```

Comment:
clippy is also complaining about this line in CI https://buildkite.com/anza/agave/builds/15002#01938ee1-02e8-4df2-8494-112c107b5803:
```suggestion
                rewards,
```